biobesu/helper/converters.py

```diff
@@ -0,0 +1,298 @@
+                # Resets id and name for new phenotype.
+                if line.startswith(match_term):
+                    hpo_id = None
+                    hpo_name = None
+                    added = False
+                # Once a phenotype is added, skips lines until next term.
+                elif added is True:
+                    continue
+                # Sets id/name when found.
+                elif line.startswith(match_id):
+                    hpo_id = line.lstrip(match_id).strip()
+                elif line.startswith(match_name):
+                    hpo_name = line.lstrip(match_name).strip()
+
+                # If a combination of an id and a name/synonym is stored, saves it to the dictionaries.
+                # Afterwards, reset id to None so that next lines will be ignored till the next phenotype.
+                if hpo_id is not None and hpo_name is not None:
+                    self.names_by_id[hpo_id] = hpo_name
+                    self.id_by_names[hpo_name] = hpo_id
+                    added = True
+
+    def id_to_name(self, hpo_ids, include_na=False):
+        """
+        Convert a (list of) phenotype ID(s) to its/their name.
+
+        :param hpo_ids: (list of) phenotype(s) to convert
+        :type hpo_ids: str | list[str]
+        :param include_na:  replace IDs with no match with "NA" in the returned output
+        :type include_na: bool
+        :return (str, None or tuple): the converted keys
+        :rtype: str | None | tuple[list[str],set[str]]
+        """
+
+        return self.key_to_value(hpo_ids, self.names_by_id, include_na)
+
+    def name_to_id(self, hpo_names, include_na=False):
+        """
+        Convert a (list of) phenotype name(s) to its/their ID.
+        :param hpo_names: (list of) phenotype(s) to convert
+        :type hpo_names: str | list[str]
+        :param include_na: replace IDs with no match with "NA" in the returned output
+        :type include_na: bool
+        :return: the converted keys
+        :rtype: str | None | tuple[list[str],set[str]]
+        """
+
+        return self.key_to_value(hpo_names, self.id_by_names, include_na)
+
+    def id_to_phenopacket(self, phenopacket_id, phenotype_ids):
+        """
+        Convert a list of phenotype IDs to a phenopacket.
+        :param phenopacket_id: the ID to be used for the phenopacket
+        :type phenopacket_id str
+        :param phenotype_ids: the phenotypes to convert
+        :type phenotype_ids: list[str]
+        :return: a JSON-formatted phenopacket string
+        :rtype: str
+        """
+
+        # Writes opening bracket.
+        output_string = '{'
+
+        # Convert row id to phenopackets.
+        output_string += '\n\t"id": "' + phenopacket_id + '",' + \
+                         '\n\t"phenotypic_features": ['
+
+        # Converts phenotypes phenopackets.
+        for i, phenotype_id in enumerate(phenotype_ids):
+            output_string += '{' \
```

Comment:
Are we sure we want to use `.format()` for all these?

Any curly bracket used for the json output would need to be escaped:
```
'{{' \
'\n\t\t"type": {{' \
'\n\t\t\t"id": "{}",' \
'\n\t\t\t"label": "{}"' \
'\n\t\t}}' \
'\n\t}}' \
.format()
```

Not sure if this would be that beneficial as it might also make those Strings (a bit) less readable.

Reply:
Right, I agree. For those you might want to keep it this way. I really dislike the `\t\t\n`'s, but I don't see any other way to make it prettier.

Reply:
Agreed regarding that it does not look pretty. Did a quick google search and I'll have a look at rewriting the code using [json.dump()](https://docs.python.org/3/library/json.html#json.dump)